src/main/kotlin/guru/stefma/androidartifacts/MavenPublication.kt

```diff
@@ -98,6 +96,34 @@ internal fun MavenPublication.setupMetadata(
     version = project.version as String
     artifactId = extension.artifactId
     groupId = project.group as String
+
+    if (GradleVersionComparator(project.gradle.gradleVersion).betterThan("4.7")) {
+        pom.name.set(extension.name ?: project.name)
+        pom.description.set(extension.description ?: project.description)
+        pom.url.set(extension.url)
+
+        // Add deprecated license property if available
+        extension.licenseSpec?.let { license ->
+            pom.licenses {
+                it.license {
+                    it.name.set(license.name)
+                    it.url.set(license.url)
+                    it.comments.set(license.comments)
+                    it.distribution.set(license.distribution)
+                }
+            }
+        }
+    } else {
+        // TODO add meta information using `pom.withXml {  }`
```

Comment:
Do you want to fix that TODO in this PR?
Or should we keep it for now?

Reply:
let'd do this in a separate PR, might also be a great beginner task 😉